components/gamification/store/StoreItemsList.tsx

```diff
@@ -19,15 +21,26 @@ const StoreItemsList  = ({ storeItemsPage, userItemsPage, isRefreshing, onRefres
   const [purchase, { isError: isPurchaseError, error: purchaseError }] = usePurchaseItemMutation();
   useError(purchaseError);
 
-  const { add } = useNotifications();
+  const [purchasingItemId, setPurchasingItemId] = useState('');
+
+  const { add, remove, clear } = useNotifications();
 
   const handleGemsPress = async (item: IStoreItemWithQuantity) => {
+    clear();
+
+    const purchasingNotificationId = add({ type: 'info', body: 'Purchasing...', time: 5000 });
```

Comment:
Notifying the user by adding a loader to the button is better than displaying "Purchasing..." as a notification

Reply:
We have both 😎

Reply:
Then we don't need one of them 😆. I think notification is only necessary when it is successfully completed.